### PR TITLE
Total OOP overhaul and unittesting

### PR DESCRIPTION
**Pull request by DeutscheGabanna**
Better thought-out structure of the objects:
1. Librarian() master-object as a builder of DatedEntriesGroup() objects
2. DatedEntriesGroup() as a builder for DatedEntry() objects

This approach follows the encapsulation principle and allows for chaining object calls.

On top of that, there is extensive testing skeleton added. Will expand it further.

The outputting methods (generating files) are still WIP and have not been included in this PR. The PR includes only parsing and building structured data out of CSV.

### CODE REVIEW

File: .idea/.gitignore
Comment (bgalek) [inline .idea/.gitignore:1]: https://askbartosz.gh.allegrogroup.com/#set-global-gitignore
add .idea and .vscode to your gitignore;)

File: _tests/hack.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import sys
```
Comment (bgalek) [inline _tests/hack.py]: we should not need this anywhere

File: _tests/test_dated_entry.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from unittest import TestCase
+import hack
```
Comment (bgalek) [inline _tests/test_dated_entry.py]: ?:)

File: _tests/dev-null
Comment (bgalek) [inline _tests/dev-null:1]: looks like mistake :)

Comment (DeutscheGabanna) [reply _tests/dev-null:1]: It was supposed to act like an empty file. Will delete it.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -8,6 +8,8 @@ jobs:
         python: [ 3.11, 3.12 ]
         os: [ ubuntu-latest, windows-latest ]
     runs-on: ${{ matrix.os }}
+    env: 
+      PYTHONUTF8: 1
```
Comment (pawel-ilnicki) [inline .github/workflows/test.yaml:12]: 🤦🏻 I swear I scoured stack overflow looking for a way to force UTF-8 in the runner. That was surprisingly easy... Thank you so much

Comment (DeutscheGabanna) [inline .github/workflows/test.yaml:12]: 🤦🏻 I swear I scoured stack overflow looking for a way to force UTF-8 in the runner. That was surprisingly easy... Thank you so much

**Review by bgalek — APPROVED** [review]

**Pull request merged**